animata/list/list-stack.tsx

```diff
@@ -0,0 +1,167 @@
+          <div className="flex flex-col items-center">
+            <AnimatePresence>
+              {showAll
+                ? cards.map((card, index) => (
+                    <motion.div
+                      key={card.id}
+                      custom={index}
+                      variants={revealAnimation}
+                      initial="hidden"
+                      animate="visible"
+                      exit="hidden"
+                      className="mb-3"
+                    >
+                      <CardStack
+                        id={card.id}
+                        icon={card.icon}
+                        title={card.title}
+                        location={card.location}
+                        date={card.date}
+                      />
+                    </motion.div>
+                  ))
+                : cards.map((card, index) => (
+                    <motion.div
+                      key={card.id}
+                      custom={index}
+                      variants={stackAnimation}
+                      initial="hidden"
+                      animate="visible"
+                      exit="hidden"
+                      style={{
+                        position: "absolute",
+                        transformOrigin: "top center",
+                        zIndex: cards.length - index,
+                      }}
+                    >
+                      <CardStack
+                        id={card.id}
+                        icon={card.icon}
+                        title={card.title}
+                        location={card.location}
+                        date={card.date}
+                      />
+                    </motion.div>
+                  ))}
+            </AnimatePresence>
+          </div>
+        </div>
+      </div>
+      <motion.button
+        className="min-w-1/2 mt-2 rounded-full border-x border-y border-gray-100 bg-white px-6 py-1.5 text-sm font-semibold text-zinc-800 shadow-lg"
+        whileHover={{ scale: 1.05 }}
+        whileTap={{ scale: 0.95 }}
+        onClick={toggleShow}
+        transition={{ type: "spring", stiffness: 50, damping: 20 }}
+      >
+        {showAll ? (
+          <>
+            <div className="flex flex-row items-center gap-1.5">
+              <p>Hide</p>
+              <motion.div
+                initial={{ rotate: 0 }}
+                animate={{ rotate: 180 }}
+                transition={{ duration: 0.4 }}
+              >
+                <ChevronDown />
+              </motion.div>
+            </div>
+          </>
+        ) : (
+          <>
+            <div className="flex flex-row items-center gap-1.5">
+              <p>Show All</p>
+              <motion.div
+                initial={{ rotate: 0 }}
+                animate={{ rotate: 0 }}
+                transition={{ duration: 0.4 }}
+              >
+                <ChevronDown />
+              </motion.div>
+            </div>
+          </>
+        )}
+      </motion.button>
```

Comment:
Same here, you can conditionally change the text and the rotate property of the chevron, keeping the same elements.